pkg/extensions/sync/http_handler.go

```diff
@@ -42,42 +46,64 @@ func (h *PostHandler) Handler(response http.ResponseWriter, request *http.Reques
 
 	uuid, err := guuid.NewV4()
 	if err != nil {
-		WriteData(response, http.StatusInternalServerError, err.Error())
+		WriteData(response, err.Error())
 
 		return
 	}
 
+	syncErrs := make(map[string]error)
+
 	for _, regCfg := range h.Cfg.Registries {
 		// if content not provided, don't run periodically sync
 		if len(regCfg.Content) == 0 {
-			h.Log.Info().Msgf("sync config content not configured for %s, will not run periodically sync", regCfg.URL)
+			h.Log.Info().Msgf("sync config content not configured for %s, will not run periodically sync", regCfg.URLs)
 
 			continue
 		}
 
 		// if pollInterval is not provided, don't run periodically sync
 		if regCfg.PollInterval == 0 {
-			h.Log.Warn().Msgf("sync config PollInterval not configured for %s, will not run periodically sync", regCfg.URL)
+			h.Log.Warn().Msgf("sync config PollInterval not configured for %s, will not run periodically sync", regCfg.URLs)
 
 			continue
 		}
 
-		upstreamRegistryName := strings.Replace(strings.Replace(regCfg.URL, "http://", "", 1), "https://", "", 1)
-
-		if err := syncRegistry(regCfg, h.StoreController, h.Log, localCtx, policyCtx,
-			credentialsFile[upstreamRegistryName], uuid.String()); err != nil {
-			h.Log.Err(err).Msg("sync http handler: error while syncing in")
-			WriteData(response, http.StatusInternalServerError, err.Error())
-
-			return
+		for _, upstreamURL := range regCfg.URLs {
+			upstreamAddr := StripRegistryTransport(upstreamURL)
+			if err = syncRegistry(regCfg, upstreamURL, h.StoreController, localCtx, policyCtx,
+				credentialsFile[upstreamAddr], uuid.String(), h.Log); err != nil {
+				h.Log.Error().Err(err).Str("registry", upstreamAddr).
+					Msg("sync http handler: error while syncing in - trying fallback registry")
+
+				syncErrs[upstreamAddr] = err
+			} else {
+				_, ok := syncErrs[upstreamAddr]
+				if ok {
+					// if fallback registry is success remove error
+					delete(syncErrs, upstreamAddr)
+				}
+
+				break
+			}
 		}
 	}
 
-	WriteData(response, http.StatusOK, "")
+	msg := ""
+	for _, err := range syncErrs {
```

Comment:
Maybe I did not get it but imagine you have 3 servers  in URLs: first 2 are not accessible and the 3rd one is. My understanding from code below will be that we will send to the client 503 error code (see line 103). Am I wrong?